File: behavioral/python/mediator/Telephone.py
from Device   import Device

class Telephone(Device):

    def __init__(self, mediator):
        Device.__init__(self, mediator)
        self._mediator.setTelephone(self)

    def runTest(self) ->None:
        print("Telephone test started.")
        self._mediator.runTelephone()

    def endTest(self) ->None:
        print("Telephone test over.")

File: behavioral/Java/mediator/src/Telephone.java
public class Telephone extends Device{
    public Telephone(Mediator mediator) {
        super(mediator);

    }

    @Override
    public void runTest() {
        System.out.println("Telephone test started.");
    }

    @Override
    public void endTest() {
        System.out.println("Telephone test over.");
    }
}
